api.go

```diff
@@ -185,21 +185,21 @@ func getDomainParams(r *http.Request, domain string) (db.DomainData, error) {
 	}
 	domainData.Email = email
 
-	domainData.MXs = r.PostForm["hostnames"]
+	for _, hostname := range r.PostForm["hostnames"] {
+		if len(hostname) == 0 {
+			continue
+		}
+		if !validDomainName(strings.TrimLeft(hostname, ".")) {
```

Comment:
Maybe TrimPrefix, since `...example.com` isn't valid, but `.example.com` is

Reply:
Good point!